be1-go/hub/standard_hub/mod.go

```diff
@@ -198,12 +174,10 @@ func (h *Hub) Start() {
 					}
 				}()
 			case id := <-h.closedSockets:
-				h.RLock()
-				for _, channel := range h.channelByID {
+				for _, channel := range h.channelByID.GetAll() {
```

Comment:
This means that technically if a channel is added while this is running, we won't unsubscribe.
Is there a mechanism to protect against this ? (just asking :-)) 

Otherwise you could also let `channelByID` manage the mutex and instead implement a ForEach function, such that you can call this like: 

```
h.channelByID.ForEach(func(c Channel){
  c.Unsubscribe(id, method.Unsubscribe{})
})
```

and ForEach could be holding the lock :)